lib/src/main/java/xyz/gianlu/librespot/audio/cdn/CdnFeedHelper.java

```diff
@@ -46,7 +46,13 @@ private CdnFeedHelper() {
 
     @NotNull
     private static HttpUrl getUrl(@NotNull Session session, @NotNull StorageResolveResponse resp) {
-        return HttpUrl.get(resp.getCdnurl(session.random().nextInt(resp.getCdnurlCount())));
+        String selectedUrl = resp.getCdnurl(session.random().nextInt(resp.getCdnurlCount()));
+        while (selectedUrl.contains("audio4-gm-fb")) {
+            LOGGER.warn("getUrl picked CDN with known issues {} (forcing re-selection)", selectedUrl );
+            selectedUrl = resp.getCdnurl(session.random().nextInt(resp.getCdnurlCount()));
+        }
+        return HttpUrl.get(selectedUrl);
+        // return HttpUrl.get(resp.getCdnurl(session.random().nextInt(resp.getCdnurlCount())));
```

Comment:
Please remove this commented line

Reply:
This is intended to only be a temporary fix - hardcoded server names and all that.  Once Spotify IT does their job correctly, it would be better to just wipe my code and go back to the previous.  I left this in there to make it easy to rip the band-aid back off.

Reply:
Yeah that's the plan, but we have git for that, perhaps you can put in a FIXME about that.